spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestChangelogTable.java

```diff
@@ -137,6 +138,64 @@ public void testOverwrites() {
         changelogRecords(snap2, snap3));
   }
 
+  @Test
+  public void testQueryWithTimeRange() {
+    sql(
+        "CREATE TABLE %s (id INT, data STRING) "
+            + "USING iceberg "
+            + "PARTITIONED BY (data) "
+            + "TBLPROPERTIES ( "
+            + " '%s' = '%d' "
+            + ")",
+        tableName, FORMAT_VERSION, formatVersion);
+
+    sql("INSERT INTO %s VALUES (1, 'a')", tableName);
+    sql("INSERT INTO %s VALUES (2, 'b')", tableName);
+
+    Table table = validationCatalog.loadTable(tableIdent);
+
+    Snapshot snap2 = table.currentSnapshot();
+
+    sql("INSERT OVERWRITE %s VALUES (-2, 'b')", tableName);
+
+    table.refresh();
+
+    Snapshot snap3 = table.currentSnapshot();
+
+    assertEquals(
+        "Should have expected changed rows only from snapshot 3",
+        ImmutableList.of(
```

Comment:
Another way to make this test a bit clearer I would probably make
```
Snapshot1Rows = ImmutableList.of(row(....
Snapshot2Rows = ...
```
then add on a helper
```
rowsWithValue(Snapshot1Rows, snap1.snapshotID)
```


Then this test kind of goes like
```
insertRows(Snapshot1Rows)
insertRows(Snapshot2Rows)
insertRows(Snapshot3Rows)

check(expected = rowsWithValue(Snapshot1Rows, snap1.snapshotID)
```

Reply:
Tried it. We have to change the `_change_ordinal` as well. Need a bit more logic to make it work, it may not be worth considering there only about 3 test cases using that. WDYT?